lib/server/socket.exs

```diff
@@ -0,0 +1,60 @@
+Code.require_file "../helpers/process_commands.exs", __DIR__
+
+defmodule Alchemist.Server.Socket do
+  use Application
```

Comment:
When I wanted to mix gen_tcp and process in OTP I had two options:
1. handle it like `lib/server/io.exs`, which means run a GenServer and accept new connections in `handle_info(:timeout, state)` which I didn't find it nice.
2. While I was looking at docs to see how people implement this, I came across [Task and gen_tcp](http://elixir-lang.org/getting-started/mix-otp/task-and-gen-tcp.html) in elixir getting started documents.

Option 2 looks decent to me, it also handles corner cases. That's why I choose this approach.

If you have other suggestion, I'd be happy to dig in and implement. Then we can compare and choose the best option